src/main/java/com/iexec/sms/tee/session/scone/SconeSessionMakerService.java

```diff
@@ -0,0 +1,161 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.sms.tee.session.scone;
+
+import com.iexec.common.tee.TeeEnclaveProvider;
+import com.iexec.sms.api.config.TeeServicesProperties;
+import com.iexec.sms.tee.ConditionalOnTeeProvider;
+import com.iexec.sms.tee.session.base.SecretEnclaveBase;
+import com.iexec.sms.tee.session.base.SecretSessionBase;
+import com.iexec.sms.tee.session.base.SecretSessionBaseService;
+import com.iexec.sms.tee.session.generic.TeeSessionGenerationException;
+import com.iexec.sms.tee.session.generic.TeeSessionRequest;
+import com.iexec.sms.tee.session.scone.cas.SconeEnclave;
+import com.iexec.sms.tee.session.scone.cas.SconeSession;
+import com.iexec.sms.tee.session.scone.cas.SconeSession.AccessPolicy;
+import com.iexec.sms.tee.session.scone.cas.SconeSession.Image;
+import com.iexec.sms.tee.session.scone.cas.SconeSession.Image.Volume;
+import com.iexec.sms.tee.session.scone.cas.SconeSession.Security;
+import com.iexec.sms.tee.session.scone.cas.SconeSession.Volumes;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+//TODO Rename and move
+@Slf4j
+@Service
+@ConditionalOnTeeProvider(providers = TeeEnclaveProvider.SCONE)
+public class SconeSessionMakerService {
+
+    // Internal values required for setting up a palaemon session
+    // Generic
+    static final String TOLERATED_INSECURE_OPTIONS = "TOLERATED_INSECURE_OPTIONS";
+    static final String IGNORED_SGX_ADVISORIES = "IGNORED_SGX_ADVISORIES";
+    static final String APP_ARGS = "APP_ARGS";
+
+    // PreCompute
+    static final String PRE_COMPUTE_ENTRYPOINT = "PRE_COMPUTE_ENTRYPOINT";
+    // PostCompute
+    static final String POST_COMPUTE_ENTRYPOINT = "POST_COMPUTE_ENTRYPOINT";
+
+    private final SecretSessionBaseService secretSessionBaseService;
+    private final TeeServicesProperties teeServicesConfig;
+    private final SconeSessionSecurityConfig attestationSecurityConfig;
+
+    public SconeSessionMakerService(
+            SecretSessionBaseService secretSessionBaseService,
+            TeeServicesProperties teeServicesConfig,
+            SconeSessionSecurityConfig attestationSecurityConfig) {
+        this.secretSessionBaseService = secretSessionBaseService;
+        this.teeServicesConfig = teeServicesConfig;
+        this.attestationSecurityConfig = attestationSecurityConfig;
+    }
+
+    /**
+     * Collect tokens required for different compute stages (pre, in, post)
+     * and build the yaml config of the TEE session.
+     * <p>
+     * TODO: Read onchain available infos from enclave instead of copying
+     * public vars to palaemon.yml. It needs ssl call from enclave to eth
+     * node (only ethereum node address required inside palaemon.yml)
+     *
+     * @param request session request details
+     * @return session config in yaml string format
+     */
+    @NonNull
```

Comment:
Is this true only for Scone and not for Gramine ?
See GramineSessionMakerService